mealie/services/event_bus_service/event_bus_service.py

```diff
@@ -23,17 +26,32 @@ def publisher(self) -> PublisherLike:
     def get_urls(self, event_type: EventTypes) -> list[str]:
         repos = AllRepositories(self.session)
 
-        notifiers: list[GroupEventNotifierPrivate] = repos.group_event_notifier.by_group(self.group_id).multi_query(
-            {"enabled": True}, override_schema=GroupEventNotifierPrivate
-        )
+        notifiers: list[GroupEventNotifierPrivate] = repos.group_event_notifier.by_group(  # type: ignore
+            self.group_id
+        ).multi_query({"enabled": True}, override_schema=GroupEventNotifierPrivate)
 
         return [notifier.apprise_url for notifier in notifiers if getattr(notifier.options, event_type.name)]
 
-    def dispatch(self, group_id: UUID4, event_type: EventTypes, msg: str = "") -> None:
-        self.group_id = group_id
+    def dispatch(self, group_id: UUID4, event_type: EventTypes, msg: str = "", event_source: dict = None) -> None:
+        self.group_id = group_id  # type: ignore
+
+        if event_source is None:
+            event_source = {}
+
+        self.event_source = event_source
 
         def _dispatch():
             if urls := self.get_urls(event_type):
+                if self.event_source:
+                    urls = [
+                        # We use query params to add custom key: value pairs to the Apprise payload by prepending the key with ":".
+                        # As of 2022-11-06 this is undocumented, but discussed in this pull request:
+                        # https://github.com/caronc/apprise/pull/547
+                        EventBusService.merge_query_parameters(url, {f":{k}": v for k, v in self.event_source.items()})
+                        for url in urls
+                        if EventBusService.is_json_url(url)
+                    ]
+
```

Comment:
Won't this filter out any non json_urls? Was that intended behavior here?

Reply:
Yeah, custom key value pairs are only supported by the custom form, custom XML, and custom JSON endpoints. The original implementation used custom headers which was only supported by JSON, but I can add XML and form endpoints and leave a note explaining why.

Reply:
I noticed that limitation too, kind of a bummer. I think the problem here is that if the is_json_url, function returns false we discard the url all together. Wouldn't that break regular Apprise integrations like discord where we're implementing them as just a message service? If we specify an Apprise URL for discord AND the event_source is defined, we'll never make the call. 

Or am I reading this wrong?

Reply:
Oh wow I totally missed that, you're right. I was only testing the JSON ones. I can an else statement so we keep all URLs